arbor/include/arbor/mechanism.hpp

```diff
@@ -60,6 +62,8 @@ class mechanism {
     // Non-global parameters can be set post-instantiation:
     virtual void set_parameter(const std::string& key, const std::vector<fvm_value_type>& values) = 0;
 
+    virtual void nrn_coalesce_init() = 0;
+
```

Comment:
I'm not sure I understand: 
* We keep the nrn_init() function in the generated mechanisms the same, but we change the name of arb::mechanism::nrn_init() to something like reset() or initialize().
Do you mean to change arb::mechanism::nrn_coalesce_init() to reset() or initialize()?

Reply:
Exactly that. `nrn_init()` would then be a virtual method in the mechanism backend classes rather than in the top-level mechanism class. I can make a ticket for changing the other names, too.